### PR TITLE
New changes and improvemenets

### PR DESCRIPTION
**Pull request by freshtechtips**
I should have made more changes before I sent this pull request but I couldn't help myself

but at least it's not a lot of changes

Improvements:
- Delete old todo and unneeded
- Improve performance by using const + dispose all the text editing controllers in dispose()
- use (Attribute.margin, Attribute.alignment) in flutter_quill_extensions
- Remove unused code
- Using a better default value for linkRegExp for inserting the images
- Update the TextFields to automatically send the form when press enter in the last field
- Update the text input types of the text fields, suggestions, and more
- Fix some MediaButton bugs and format code
- Format and organize some of the source code, even in the tests, some of them done in `dart format .`
- Fix bug [#1432](https://github.com/singerdmx/flutter-quill/issues/1432)
- Fix bug [#1434](https://github.com/singerdmx/flutter-quill/issues/1434)
- Fix bug [#1435](https://github.com/singerdmx/flutter-quill/issues/1435) by sending [pull request](https://github.com/bdlukaa/fluent_ui/pull/946) to fluent_ui maintainers, in case they decided to reject the changes, then you need to make sure to:

```dart
      localizationsDelegates: const [
        DefaultCupertinoLocalizations.delegate,
        DefaultMaterialLocalizations.delegate,
        DefaultWidgetsLocalizations.delegate,
        FluentLocalizations.delegate,
      ],
``` 
when using FluentApp or any custom widget app

- Add isMacOS() function in the platform checker file 

Features:
1. Add MediaButton for picking the images only since the video one is not ready
3. Thanks to feature request [#1431](https://github.com/singerdmx/flutter-quill/issues/1431) by [milindgoel](https://github.com/milindgoel15)  I was able to add a new feature which allows customizing the text selection in quill editor which is useful for custom theme design system for custom app widget like FluentApp

### CODE REVIEW

**Comment by singerdmx** [comment]
Is this PR ready to merge or you need to make more changes

File: example/android/app/build.gradle [hunk 1]
```diff
@@ -53,7 +53,7 @@ android {
 
     buildTypes {
         release {
-            // TODO: Add your own signing config for the release build.
+            // I removed the todo because it's not required for this example
```
Comment (singerdmx) [inline example/android/app/build.gradle:56]: Just remove this comment then

File: example/android/app/build.gradle [hunk 2]
```diff
@@ -42,7 +42,7 @@ android {
     }
 
     defaultConfig {
-        // TODO: Specify your own unique Application ID (https://developer.android.com/studio/build/application-id.html).
+        // I removed the todo because it's not required for this example
```
Comment (singerdmx) [inline example/android/app/build.gradle:45]: Just remove this comment then

**Review by singerdmx — CHANGES_REQUESTED** [review]

**Comment by freshtechtips** [comment]
> Is this PR ready to merge or you need to make more changes

If you merge this then I will prepare for the next pull request

**Comment by singerdmx** [comment]
Can you cherry pick the other change related to toolbar in your next PR

**Comment by freshtechtips** [comment]
> Ok

I fixed what you want

> Can you cherry pick the other change related to toolbar in your next PR

Which changes??

**Comment by singerdmx** [comment]
https://github.com/singerdmx/flutter-quill/pull/1438

**Comment by freshtechtips** [comment]
> #1438

That was not my pull request, I can carry his changes but we will have to solve the conflicts

but that is not what I'm talking about, do you want me to take the changes in the next pull request or do you want him to solve the conflicts?? everything looks good just the text selection color cursor should be removed, I don't have access to change code so I will ask him, give me a few moments

**Pull request merged**